File: frontend/src/types/warehouse/warehouseBom.ts
export interface WarehouseBom {
  id: number
  category: string
  genericName: string
  brand: string | null
  name: string
  model: string
  remark: string | null
  imageObjectKey: string | null
  imageUrl: string | null
  imageObjectKeys: string[]
  imageUrls: string[]
  createdAt: string
  updatedAt: string
}

export interface WarehouseBomQuery {
  category?: string
  genericName?: string
  brand?: string
  name?: string
  model?: string
  page?: number
  pageSize?: number
}

export interface WarehouseBomSavePayload {
  category: string
  genericName: string
  brand?: string | null
  name: string
  model?: string | null
  remark?: string | null
  imageObjectKey?: string | null
  imageObjectKeys?: string[]
}

export interface BomFilterLinkageQuery {
  category?: string
  genericName?: string
  brand?: string
}

export interface BomFilterOptions {
  categories: string[]
  genericNames: string[]
  brands: string[]
}

export type WarehouseBomExportQuery = Omit<WarehouseBomQuery, 'page' | 'pageSize'> & {
  ids?: number[]
}

File: backend/src/main/java/com/storage/warehouse/entity/WarehouseBom.java
package com.storage.warehouse.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("warehouse_bom")
public class WarehouseBom {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String category;

    private String genericName;

    private String brand;

    private String name;

    private String remark;

    private String imageObjectKey;

    @TableField(exist = false)
    private String imageUrl;

    @TableField(exist = false)
    private List<String> imageObjectKeys;

    @TableField(exist = false)
    private List<String> imageUrls;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
